x-pack/plugins/enterprise_search/public/applications/shared/types.ts

```diff
@@ -4,14 +4,7 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export interface IFlashMessagesProps {
-  info?: string[];
-  warning?: string[];
-  error?: string[];
-  success?: string[];
-  isWrapped?: boolean;
-  children?: React.ReactNode;
-}
```

Comment:
@byronhulcher @scottybollinger @JasonStoltz I wanted to highlight/get your thoughts on a data structure change I'm proposing for Flash Messages. Rather than passing in individual error/success/info/warning keys, I'd like to see us treat flash messages as an array of IFlashMessage objects that look like this:

```js
[
  {
    type: 'error', // | success | info | warning
    message: 'Something went wrong', // the main title/error text
    description: <EuiLink>Fix it here</EuiLink>, // any extra description, subtext, action, etc.
  }
]
```

This allows us to simplify our FlashMessages component to iterate through messages instead of having to check 4 different arrays.

This also allows us some flexibility in our flash messages - for example, (contributed hypothetical example) if in the future we need a certain flash message to persist across the entire site, we could then easily add a new `shouldPersist: true` boolean flag to flash messages and filter the messages array based on that property. 

Let me know if you see any potential issues with this change!

Reply:
Makes sense to me.

Reply:
No issue. Can you share an example of implementation?

Reply:
Here's a few more if that helps!

```
FlashMessagesLogic.actions.setMessage([
  { type: 'success', message: 'Successfully deleted item' }
  { type: 'warning', message: 'Some other warning', description: 'A detailed message body, you can use JSX' }
  { type: 'info', message: 'This is probably too many flash messages' }
]);
```
You can send either an array of flash message objs or a single obj if it's just one flash message. Both `setMessages` and `setQueuedMessages` work the same way.

Reply:
I like this! I think it could be useful if/when we want to start pushing rather than setting flash messages.